x-pack/plugins/session_view/server/routes/alerts_route.test.ts

```diff
@@ -0,0 +1,133 @@
+import { loggingSystemMock } from 'src/core/server/mocks';
+import { alertingAuthorizationMock } from '../../../alerting/server/authorization/alerting_authorization.mock';
+import { auditLoggerMock } from '../../../security/server/audit/mocks';
+import { AlertingAuthorizationEntity } from '../../../alerting/server';
+import { ruleDataServiceMock } from '../../../rule_registry/server/rule_data_plugin_service/rule_data_plugin_service.mock';
+
+const alertingAuthMock = alertingAuthorizationMock.create();
+const auditLogger = auditLoggerMock.create();
+
+const DEFAULT_SPACE = 'test_default_space_id';
+
+const getEmptyResponse = async () => {
+  return {
+    hits: {
+      total: 0,
+      hits: [],
+    },
+  };
+};
+
+const getResponse = async () => {
+  return {
+    hits: {
+      total: mockEvents.length,
+      hits: mockEvents.map((event) => {
+        return {
+          found: true,
+          _type: 'alert',
+          _index: '.alerts-security',
+          _id: 'NoxgpHkBqbdrfX07MqXV',
+          _version: 1,
+          _seq_no: 362,
+          _primary_term: 2,
+          _source: {
+            [ALERT_RULE_TYPE_ID]: 'apm.error_rate',
+            message: 'hello world 1',
+            [ALERT_RULE_CONSUMER]: 'apm',
+            [ALERT_WORKFLOW_STATUS]: 'open',
+            [SPACE_IDS]: ['test_default_space_id'],
+            ...event,
+          },
+        };
+      }),
+    },
+  };
+};
+
+const esClientMock = elasticsearchServiceMock.createElasticsearchClient(getResponse());
+
+const alertsClientParams: jest.Mocked<ConstructorOptions> = {
+  logger: loggingSystemMock.create().get(),
+  authorization: alertingAuthMock,
+  auditLogger,
+  ruleDataService: ruleDataServiceMock.create(),
+  esClient: esClientMock,
+};
+
+describe('alerts_route.ts', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    alertingAuthMock.getSpaceId.mockImplementation(() => DEFAULT_SPACE);
+    // @ts-expect-error
+    alertingAuthMock.getAuthorizationFilter.mockImplementation(async () =>
+      Promise.resolve({ filter: [] })
+    );
+    // @ts-expect-error
+    alertingAuthMock.getAugmentedRuleTypesWithAuthorization.mockImplementation(async () => {
+      const authorizedRuleTypes = new Set();
+      authorizedRuleTypes.add({ producer: 'apm' });
+      return Promise.resolve({ authorizedRuleTypes });
+    });
+
+    alertingAuthMock.ensureAuthorized.mockImplementation(
+      // @ts-expect-error
+      async ({
+        ruleTypeId,
+        consumer,
+        operation,
+        entity,
+      }: {
+        ruleTypeId: string;
+        consumer: string;
+        operation: string;
+        entity: typeof AlertingAuthorizationEntity.Alert;
+      }) => {
+        if (ruleTypeId === 'apm.error_rate' && consumer === 'apm') {
+          return Promise.resolve();
+        }
+        return Promise.reject(new Error(`Unauthorized for ${ruleTypeId} and ${consumer}`));
+      }
+    );
+  });
+
+  describe('doSearch(client, sessionEntityId)', () => {
```

Comment:
This test is great, but it tests the function executed inside and it doesn't test the route itself. Please check the examples [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/routes/get_rule.test.ts).
As the long term vision I suggest to split the routes and its internal logic to the library files or `SessionViewClient`, if you want to expose this login not only with the routs.